File: 2018a/2018a_lee.cpp
#include <iostream>
#include <sstream>
#include <string>
#include <unordered_set>

using namespace std;

int main() {
    
	int P,K,C,temp;
	cin >> P;
	bool right = false;
	string line;

	unordered_set<int> pots;
	for ( int i=0; i<P; i++ ) {

		right = false;
		cin >> K >> C;
		getline(cin,line);
		istringstream iss(line);
		pots.insert(0);
		while( iss >> temp ) {
			cout << temp << " ";
			for ( auto pot : pots ) {
				pots.insert(pot + temp);
				if ( pot + temp == C ) {
					right = true;
					goto end;
				}

			}
			
		}

end:
		cout << i+1 << " " << ( right ? "YES" : "NO" ) << "\n";

		pots.clear();

	}


}

File: 2018a/2018a_lee.py
P = int(input())

for i in range(P):
	cor = False
	line = input()
	K,C = [ int(x) for x in line.split()[0:2] ]
	pots = [ int(x) for x in line.split()[2:] ]

	sums = {0}
	for p in pots:
		tempSums = set()
		for s in sums:
			tempSums.add( s+p )
			if s+p == C:
				cor = True
				break
		if cor:
			break
		sums = sums.union( tempSums )

	print( str(i+1), end=" " )
	print( "YES" ) if cor else print( "NO" )

